src/module_wrap.cc

```diff
@@ -59,6 +60,9 @@ static const char* const EXTENSIONS[] = {
   ".mjs"
 };
 
+typedef std::unordered_set<std::string> ConditionSet;
+static std::unordered_map<std::string, ConditionSet> conditions_cache_;
```

Comment:
These should probably be members of the `Enviroment` class? I don’t understand the exact semantics here, but this is definitely prone to race conditions in its current state.

(That being said, I’ll try to open a PR to move all the resolve code into JS; there is no reason why we need to complicate things by using C++, imo)